build.gradle

```diff
@@ -20,15 +20,25 @@ allprojects {
     }
 
     dependencies {
+        // Scala version for all scala dependencies
+        def scalaVersion = '2.12'
+
         // Use Scala 2.12 in our library project
         implementation 'org.scala-lang:scala-library:2.12.7'
 
         // Use Scalatest for testing our library
-        testImplementation 'junit:junit:4.12'
-        testImplementation 'org.scalatest:scalatest_2.12:3.0.5'
+        testImplementation "junit:junit:4.12"
+        testImplementation "org.scalatest:scalatest_$scalaVersion:3.0.5"
 
         // Need scala-xml at test runtime
-        testRuntimeOnly 'org.scala-lang.modules:scala-xml_2.12:1.1.1'
+        testRuntimeOnly "org.scala-lang.modules:scala-xml_$scalaVersion:1.1.1"
+
+        // Logging
+        compile group: 'ch.qos.logback', name: 'logback-classic', version: '1.2.3'
```

Comment:
This should go. Libraries should only depend on SL4J or similar interfaces. Scalalogging uses SL4J.

Reply:
It’s fine to declare it as a test runtime dependency.